src/dispatch/scheduler.py

```diff
@@ -7,28 +7,26 @@
 .. moduleauthor:: Kevin Glisson <kglisson@netflix.com>
 .. moduleauthor:: Marc Vilanova <mvilanova@netflix.com>
 """
-import threading
 import logging
 import time
 
+from multiprocessing.pool import ThreadPool
```

Comment:
Without getting too deep here, you prefer this over `concurrent.futures.ThreadPoolExecutor`?